src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs

```diff
@@ -28,17 +28,28 @@ internal static partial class FileSystem
             UnixFileMode.OtherWrite |
             UnixFileMode.OtherExecute;
 
+        // TryCloneFile is defined in either FileSystem.TryCloneFile.OSX.cs or FileSystem.TryCloneFile.OtherUnix.cs.
+        private static partial bool TryCloneFile(string sourceFullPath, in Interop.Sys.FileStatus srcStat, UnixFileMode filePermissions, string destFullPath, bool overwrite);
+
         public static void CopyFile(string sourceFullPath, string destFullPath, bool overwrite)
         {
-            long fileLength;
-            UnixFileMode filePermissions;
-            using SafeFileHandle src = SafeFileHandle.OpenReadOnly(sourceFullPath, FileOptions.None, out fileLength, out filePermissions);
+            // Open the src file handle, and read the file permissions.
+            using SafeFileHandle src = SafeFileHandle.OpenReadOnly(sourceFullPath, FileOptions.None, out Interop.Sys.FileStatus srcFileStatus);
+            UnixFileMode filePermissions = SafeFileHandle.GetFileMode(srcFileStatus);
```

Comment:
Remove `SafeFileHandle.GetFileMode` and call `srcFileStatus.GetUnixFileMode` (& mask) here.